### ISSUE TITLE
Document::content_type should be of type Mime

### ISSUE THREAD

**Issue by nox** [servo/servo#16483]

**Comment by jdm** [comment]
@nupurbaghel This would be a good issue to work on. The [Mime](https://doc.servo.org/mime/struct.Mime.html) allows us to store a more structured Content-Type header value than just a string, and we can create specific values through parsing a string value or constructing them like `Mime(TopLevel::Text, SubLevel::Html, vec![])`.

**Comment by nupurbaghel** [comment]
Hi Josh,

Thanks for suggesting, I will start looking into it now itself :)

Nupur
Team WeCode

**Comment by nupurbaghel** [comment]
@jdm Sure. I will like to take up this issue. It sounds quite interesting :)

**Comment by nox** [comment]
The `Mime` type with three separate fields is gone from the latest `mime` crate, I wonder if we should wait for that bump before doing this change.

**Comment by jdm** [comment]
I don't see why waiting is a benefit.

**Comment by nupurbaghel** [comment]
According to my understanding of the file document.rs, the places where Mime type could be included are :
1. CreateElement function : where we check for self.content_type == "application/xhtml+xml”
2. The document object created in new_inherited function which includes : 
 IsHTMLDocument::**HTMLDocument => "text/html"** and
 IsHTMLDocument::**NonHTMLDocument => "application/xml"**

But, since other files still use the content_type as DOMString, we decided to not change the new and new_inherited functions and instead only update direct matching of strings with parse() function. So, this leaves us with just the CreateElement function. Are there any other functions that I am missing out? And should I proceed with updating the CreateElement function for now?
